### PR TITLE
start trying to snip links between kube-apiserver and openshift-apiserver

### PR DESCRIPTION
**Pull request by deads2k**
I'm trying to start localizing kube-apiserver injected code without building a staging repo for it since it probably won't land in staging.  It's a little unclear how to handle cross deps with origin, but at least we have a starting point for separating the mess.

@sttts who wants in on this one?

### CODE REVIEW

File: pkg/authorization/apiserver/registry/subjectaccessreview/rest.go [hunk 1]
```diff
@@ -165,3 +164,12 @@ func (r *REST) isAllowed(user user.Info, sar *authorizationapi.SubjectAccessRevi
 
 	return nil
 }
+
+// isPersonalAccessReviewFromSAR this variant handles the case where we have an SAR
+func isPersonalAccessReviewFromSAR(sar *authorizationapi.SubjectAccessReview) bool {
+	if len(sar.User) == 0 && len(sar.Groups) == 0 {
```
Comment (mfojtik) [inline pkg/authorization/apiserver/registry/subjectaccessreview/rest.go]: nit:

```
return len(sar.User) == 0 && len(sar.Groups) == 0
```

File: test/extended/oauth/well_known.go [hunk 2]
```diff
@@ -6,13 +6,14 @@ import (
 	"net/url"
 	"path"
 
+	"github.com/openshift/library-go/pkg/oauth/oauthdiscovery"
```
Comment (mfojtik) [inline test/extended/oauth/well_known.go]: nit: move down :-)

**Comment by deads2k** [comment]
nits addressed in https://github.com/openshift/origin/pull/23153

if this is tagged, it's a good chance to instant-merge

**Pull request merged**